### PR TITLE
docs:add two ecosystem blogs

### PR DESCRIPTION
**Pull request by KID-G**
Changes:

Add two ecosystem blogs

### CODE REVIEW

File: blog/zh/blog/2022/11/23/why-is-not-reload-hot-loaded-in-nginx.md [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+使用 APISIX 你不需要重启服务就可以更新配置，这意味着修改上游、路由、插件时都不用重启。既然是基于 NGINX，APISIX 又是如何摆脱 NGINX 的限制实现完美热更新？我们先看下 APISIX 的架构。
+
+![APISIX 架构图](https://static.apiseven.com/2022/10/09/6341c29869628.png)
+
+通过上述架构图可以看到，之所以 APISIX 能摆脱 NGINX 的限制是因为它把上游等配置全部放到 APISIX Core 和 Plugin Runtime 中动态指定。
+
+以路由为例，NGINX 需要在配置文件内进行配置，每次更改都需要 reload 之后才能生效。而为了实现路由动态配置，Apache APISIX 在 NGINX 配置文件内配置了单个 server，这个 server 中只有一个 location。我们把这个 location 作为主入口，所有的请求都会经过这个 location，再由 APISIX Core 动态指定具体上游。因此 Apache APISIX 的路由模块支持在运行时增减、修改和删除路由，实现了动态加载。所有的这些变化，对客户端都零感知，没有任何影响。
+再来几个典型场景的描述。
+
+比如增加某个新域名的反向代理，在 APISIX 中只需创建上游，并添加新的路由即可，整个过程中不需要 NGINX 进程有任何重启。再比如插件系统，APISIX 可以通过 ip-restriction 插件实现 IP 黑白名单功能，这些能力的更新也是动态方式，同样不需要重启服务。借助架构内的 ETCD，配置策略以增量方式实时推送，最终让所有规则实时、动态的生效，为用户带来极致体验。
```
Comment (SylviaBABY) [inline blog/zh/blog/2022/11/23/why-is-not-reload-hot-loaded-in-nginx.md]:
```suggestion
比如增加某个新域名的反向代理，在 APISIX 中只需创建上游，并添加新的路由即可，整个过程中不需要 NGINX 进程有任何重启。再比如插件系统，APISIX 可以通过 `ip-restriction` 插件实现 IP 黑白名单功能，这些能力的更新也是动态方式，同样不需要重启服务。借助架构内的 etcd，配置策略以增量方式实时推送，最终让所有规则实时、动态的生效，为用户带来极致体验。
```

File: blog/zh/blog/2022/11/23/why-is-not-reload-hot-loaded-in-nginx.md [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+
+如果你的业务变更频率是每周或者每天，那么 NGINX 这种 reload 还是满足你的需求的。但如果变更频率是每小时、每分钟呢？假设你有 100 个 NGINX 服务，每小时 reload 一次的话，就要 reload 2400 次；如果每分钟 reload 一次，就是 864 万次。这显然是无法接受的。
+
+因此，我们需要一个不需要进程替换的 reload 方案，在现有 NGINX 进程内可以直接完成内容的更新和实时生效。
+
+## 在内存中直接生效的热加载方案
+
+在 Apache APISIX 诞生之初，就是希望来解决 NGINX 热加载这个问题的。
+
+Apache APISIX 是基于 NGINX + Lua 的技术栈，以  ETCD 作为配置中心实现的云原生、高性能、全动态的微服务 API 网关，提供负载均衡、动态上游、灰度发布、精细化路由、限流限速、服务降级、服务熔断、身份认证、可观测性等数百项功能。
```
Comment (SylviaBABY) [inline blog/zh/blog/2022/11/23/why-is-not-reload-hot-loaded-in-nginx.md]:
```suggestion
Apache APISIX 是基于 NGINX + Lua 的技术栈，以  etcd 作为配置中心实现的云原生、高性能、全动态的微服务 API 网关，提供负载均衡、动态上游、灰度发布、精细化路由、限流限速、服务降级、服务熔断、身份认证、可观测性等数百项功能。
```

File: blog/zh/blog/2022/11/25/how-apisix-support-1000-pods.md [hunk 3]
```diff
@@ -0,0 +1,409 @@
...
+
+EndpointSlice 旨在通过分片的方式来解决此问题，并没有使用单个 Endpoints 资源跟踪服务的所有网络端点，而是将它们拆分为多个较小的 EndpointSlice。
+
+默认情况下，控制面创建和管理的 EndpointSlice 将包含不超过 100 个端点。 你可以使用 [kube-controller-manager](https://kubernetes.io/zh-cn/docs/reference/command-line-tools-reference/kube-controller-manager/) 的 `--max-endpoints-per-slice` 标志设置此值，其最大值为 1000。
+
+### 为什么需要 Endpointslices？
+
+首先，我们考虑具有 2000 个 Pod 的服务它最终可能具有 1.0 MB 的 Endpoints 资源。在生产环境中，如果该服务发生滚动更新或节点迁移，那么 Endpoints 资源将会频繁变更。
+
+想象一下，如果滚动更新会导致全部 Pod 都被替换，[由于 etcd 具有最大请求大小限制](https://etcd.io/docs/v3.3/dev-guide/limit/#request-size-limit)，Kubernets 对 Endpoints 最大容量限制为 1000，如果网络端点数量超出了 1000，那么多出来的网络端点，将不会被 Endpoints 资源记录。
```
Comment (SylviaBABY) [inline blog/zh/blog/2022/11/25/how-apisix-support-1000-pods.md]:
```suggestion
想象一下，如果滚动更新会导致全部 Pod 都被替换，[由于 etcd 具有最大请求大小限制](https://etcd.io/docs/v3.3/dev-guide/limit/#request-size-limit)，Kubernetes 对 Endpoints 最大容量限制为 1000，如果网络端点数量超出了 1000，那么多出来的网络端点，将不会被 Endpoints 资源记录。
```

File: blog/zh/blog/2022/11/25/how-apisix-support-1000-pods.md [hunk 4]
```diff
@@ -0,0 +1,409 @@
...
+  namespace: default
+ports:
+- name: http
+  port: 80
+  protocol: TCP
+```
+
+## 总结
+
+本文介绍了 Kubnertes 需要部署大量 Pod 的场景和遇到的问题，对比了 Endpoints 和 Endpointslice 之间区别，以及如何安装 APISIX Ingress 来使用 Endpointslice 的特性。
```
Comment (SylviaBABY) [inline blog/zh/blog/2022/11/25/how-apisix-support-1000-pods.md]:
```suggestion
本文介绍了 Kubernetes 需要部署大量 Pod 的场景和遇到的问题，对比了 Endpoints 和 Endpointslice 之间区别，以及如何安装 APISIX Ingress 来使用 Endpointslice 的特性。
```

File: blog/zh/blog/2022/11/25/how-apisix-support-1000-pods.md [hunk 5]
```diff
@@ -0,0 +1,409 @@
...
+
+但随着 Kubernetes 集群和服务逐渐开始为更多的后端 Pod 进行处理和发送请求，比如上文提到大流量场景下，Pod 数量会被不断扩容，Endpoints API 也将变得越大。这种情况下，Endpoints API 局限性变得越来越明显，甚至成为性能瓶颈。
+
+为了解决这个局限性问题，在 Kubernetes v1.21 的版本中引入了对 Endpointslice API 的支持，解决了 Endpoints API 处理大量网络端点带来的性能问题，同时提供了可扩展和可伸缩的能力。
+
+通过下图我们可以明显看到它们之间的区别：
+
+- Endpoints 在流量高峰时的变化：
+
+![Endpoints 在流量高峰时的变化](https://static.apiseven.com/2022/11/25/638039cf30502.png)
```
Comment (SylviaBABY) [inline blog/zh/blog/2022/11/25/how-apisix-support-1000-pods.md:36]: maybe we can add a [tab] to keep this section indented consistently
image

**Review by SylviaBABY — APPROVED** [review]

**Review by bzp2010 — APPROVED** [review]

**Pull request merged**